src/DOOM/DOOM.c

```diff
@@ -629,35 +629,22 @@ const unsigned char* doom_get_framebuffer(int channels)
     {
         return screen_buffer;
     }
-    else if (channels == 3)
+    else if (channels == 3 || channels == 4)
     {
         for (i = 0, len = SCREENWIDTH * SCREENHEIGHT; i < len; ++i)
         {
-            int k = i * 3;
+            int k = i * channels;
             int kpal = screen_buffer[i] * 3;
             final_screen_buffer[k + 0] = screen_palette[kpal + 0];
             final_screen_buffer[k + 1] = screen_palette[kpal + 1];
             final_screen_buffer[k + 2] = screen_palette[kpal + 2];
+
+            if (channels == 4)
+               final_screen_buffer[k + 3] = 255;
         }
         return final_screen_buffer;
     }
-    else if (channels == 4)
-    {
-        for (i = 0, len = SCREENWIDTH * SCREENHEIGHT; i < len; ++i)
-        {
-            int k = i * 4;
-            int kpal = screen_buffer[i] * 3;
-            final_screen_buffer[k + 0] = screen_palette[kpal + 0];
-            final_screen_buffer[k + 1] = screen_palette[kpal + 1];
-            final_screen_buffer[k + 2] = screen_palette[kpal + 2];
-            final_screen_buffer[k + 3] = 255;
-        }
-        return final_screen_buffer;
-    }
```

Comment:
Put back the duplicated code. Good thinking though. But it's ok to duplicate code that have difference, especially small block like this. Some CPUs will have the branch prediction, but some others like on micro controllers, might now. Reminder this is to be run on anything. I prefer keeping it separate and branch-less if it doesnt have to.

The 2 other changes are good. (The return 0 and the readme fix)
Rebase & Squash

Reply:
That was the whole point of this pull request

Reply:
You're probably right when it comes to the CPU. But also I'd think this is something that would  be caught by the compiler first.